scripts/processAndDownloadContracts.ts

```diff
@@ -16,28 +16,20 @@ async function processContracts() {
 			...await getMiscAddresses(),
 			...await getAaveV2Misc()
 		]
-	const output = `
-import * as contractData from './contractMetadata.json';
-export type ContractDefinition = {
-	name: string,
-	logoUri?: string,
-	protocol?: string,
-}
-export const contractMetadata = new Map<string, ContractDefinition>(
-	contractData.reduce(( acc, [address, name, protocol, logoUri] ) => {
-		if (!address || !name) return acc
-		return acc.concat([
-			[address, {
-				name: name,
-				...protocol ? {protocol} : {},
-				...logoUri ? {logoUri} : {},
-			}]])
-	}, [] as [string, ContractDefinition][])
-)
-`
-	const jsonData = JSON.stringify(contractData.map(( x) => [addressString(x.address), x.data.name, x.data.protocol, x.data.logoUri]), null, '\t')
-	fs.writeFileSync(`${OUTPUT_SRC_DIR}/contractMetadata.json`, jsonData, 'utf-8')
-	fs.writeFileSync(`${OUTPUT_SRC_DIR}/contractMetadata.ts`, output, 'utf-8')
+
+	const jsonData = JSON.stringify(contractData.map(( x) => [addressString(x.address), x.data.name, x.data.protocol, ...'logoUri' in x.data ? [x.data.logoUri] : []]), null, '\t')
+	const tsJsonData = `
+export type Address = \`0x$\{string}\`
+export type Name = string
+export type Protocol = 'Uniswap' | 'Uniswap V2' | 'Uniswap V3' | 'ETH2' | 'Arbitrum' | 'Lido' | 'Matic' | 'OpenSea' | '0x' | 'Zapper' | 'Aave'
+export type LogoRelativePath = string
+export type ContractMetadataWithLogo = readonly [Address, Name, Protocol, LogoRelativePath]
+export type ContractMetadataWithoutLogo = readonly [Address, Name, Protocol]
+
+export type ContractMetadata = readonly (ContractMetadataWithLogo | ContractMetadataWithoutLogo)[]
```

Comment:
Should `ContractMetadatas` be plural, since it is referring to a collection of `ContractMetadata` tuples?  Alternatively, `ContractMetadataCollection` or `ContractMetadataList`.

Reply:
Below you used `MetadataData`, perhaps `Metadata` refers to the collection, which means a single entry would be `ContractMetadataData` (so we should rename above two types to `ContractMetadataDataWithLogo` and `ContractMetadataDataWithoutLogo`?